ext/intl/resourcebundle/resourcebundle_iterator.c

```diff
@@ -105,7 +105,11 @@ static void resourcebundle_iterator_key( zend_object_iterator *iter, zval *key )
 	}
 
 	if (iterator->is_table) {
-		ZVAL_STRING(key, iterator->currentkey);
+		if (EXPECTED(iterator->currentkey)) {
+			ZVAL_STRING(key, iterator->currentkey);
+		} else {
+			ZVAL_NULL(key);
+		}
```

Comment:
That doesn't look quite right, since the corresponding *value* would be `Z_ISUNDEF` and we (usually) don't expose these.

I have no idea, though, what to do instead when `resourcebundle_iterator_read()` fails with `U_MISSING_RESOURCE_ERROR`. Hmm, since `(new ResourceBundle('', NULL))->get('calendar')->get('buddhist')->get(3)` returns `null`, maybe we should also set the value to `null`.